File: packages/javascript/lib/word/randPerson.ts
import type { RandThemedWordOptions, WordDetail } from '../_types/global.js';
import { themedWord } from './wordGenerator.js';

/**
 * People by age, kinship and character, never by trade and never by name.
 *
 * `randWord({ theme: 'person' })` with the theme already chosen; every other
 * option is the same.
 *
 * @example
 * randPerson({ language: 'ko', count: 3 }); // ['꼬마', '이웃', '손님']
 * randPerson({ language: 'en', count: 3 }); // ['Toddler', 'Neighbor', 'Stranger']
 */
export function randPerson(options?: RandThemedWordOptions & { output?: 'value' }): string[];
/**
 * Person words with the language and theme each one came from.
 *
 * @example
 * randPerson({ language: 'ko', output: 'detail' });
 * // [{ word: '꼬마', language: 'ko', theme: 'person' }]
 */
export function randPerson(options: RandThemedWordOptions & { output: 'detail' }): WordDetail[];
export function randPerson(options: RandThemedWordOptions = {}): string[] | WordDetail[] {
	return themedWord('person', options);
}

File: packages/dart/lib/src/word/rand_person.dart
import 'dart:math';

import 'package:randino/src/types.dart';
import 'package:randino/src/word/rand_word.dart';

/// People by age, kinship and character, never by trade and never by name.
///
/// [randWord] with the theme already chosen; every other parameter is the same.
/// For the detail form, pass [WordTheme.person] to `randWordDetails` — Dart has
/// no overloads, and a second function per theme would be a second wall of them.
///
/// ```dart
/// randPerson(language: WordLanguage.ko, count: 3); // [꼬마, 이웃, 손님]
/// randPerson(language: WordLanguage.en, count: 3); // [Toddler, Neighbor, Stranger]
/// ```
List<String> randPerson({
  WordLanguage? language,
  int count = 1,
  RandRealism realism = RandRealism.real,
  RandVocabulary vocabulary = RandVocabulary.full,
  int? minLength,
  int? maxLength,
  String? startsWith,
  bool unique = false,

  /// Where the randomness comes from: `Random.secure()` for a value nobody may
  /// predict, `Random(42)` for one that has to come out the same every run.
  Random? random,
}) => randWord(
  language: language,
  theme: WordTheme.person,
  count: count,
  realism: realism,
  vocabulary: vocabulary,
  minLength: minLength,
  maxLength: maxLength,
  startsWith: startsWith,
  unique: unique,
  random: random,
);
